proxy/coco/src/state.rs

```diff
@@ -0,0 +1,733 @@
+    pub fn with_browser<F, T>(&self, urn: &RadUrn, callback: F) -> Result<T, Error>
+    where
+        F: Send + FnOnce(&mut git::Browser) -> Result<T, Error>,
+    {
+        let git_dir = self.monorepo();
+
+        let project = self.get_project(urn, None)?;
+        let default_branch = git::Branch::local(project.default_branch());
+        let repo = git::Repository::new(git_dir)?;
+        let namespace = git::Namespace::try_from(project.urn().id.to_string().as_str())?;
+        let mut browser = git::Browser::new_with_namespace(&repo, &namespace, default_branch)?;
+
+        callback(&mut browser)
+    }
+
+    /// Initialize a [`librad_project::Project`] that is owned by the `owner`.
+    /// This kicks off the history of the project, tracked by `librad`'s mono-repo.
+    ///
+    /// # Errors
+    ///
+    /// Will error if:
+    ///     * The signing of the project metadata fails.
+    ///     * The interaction with `librad` [`librad::git::storage::Storage`] fails.
+    pub fn init_project<P: AsRef<path::Path> + Send>(
+        &self,
+        signer: &signer::BoxedSigner,
+        owner: &User,
+        project: &project::Create<P>,
+    ) -> Result<librad_project::Project<entity::Draft>, Error> {
+        let storage = self.api.storage().reopen()?;
+
+        let mut meta = project.build(owner, signer.public_key().into())?;
+        meta.sign_owned(signer)?;
+
+        let urn = meta.urn();
+        if storage.has_urn(&urn)? {
+            return Err(Error::EntityExists(urn));
+        }
+
+        let repo = storage.create_repo(&meta)?;
+        repo.set_rad_self(librad::git::storage::RadSelfSpec::Urn(owner.urn()))?;
+        log::debug!("Created project with Urn '{}'", urn);
+
+        let repo = project.setup_repo(LocalUrl::from_urn(urn, self.api.peer_id().clone()))?;
+        log::debug!("Setup repository at path '{}'", repo.path().display());
+
+        Ok(meta)
+    }
+
+    /// Create a [`user::User`] with the provided `handle`. This assumes that you are creating a
+    /// user that uses the secret key the `PeerApi` was configured with.
+    ///
+    /// # Errors
+    ///
+    /// Will error if:
+    ///     * The signing of the user metadata fails.
+    ///     * The interaction with `librad` [`librad::git::storage::Storage`] fails.
+    pub fn init_user(
+        &self,
+        signer: &signer::BoxedSigner,
+        handle: &str,
+    ) -> Result<user::User<entity::Draft>, Error> {
+        // Create the project meta
+        let mut user =
+            user::User::<entity::Draft>::create(handle.to_string(), signer.public_key().into())?;
+        user.sign_owned(signer)?;
+        let urn = user.urn();
+
+        // Initialising user in the storage.
+        {
+            let storage = self.api.storage().reopen()?;
+
+            if storage.has_urn(&urn)? {
+                return Err(Error::EntityExists(urn));
+            } else {
+                let _repo = storage.create_repo(&user)?;
+            }
+        }
+
```

Comment:
```suggestion
        // Initialise the user in the storage
        if self.api.storage().reopen()?.has_urn(&urn)? {
            return Err(Error::EntityExists(urn));
        } else {
            let _repo = storage.create_repo(&user)?;
        }

```

Reply:
In your proposal `storage` won't be available in the else branch.